tests/test_cache_lock_file_perf.py

```diff
@@ -0,0 +1,110 @@
+import multiprocessing
+import threading
+import time
+from msal_extensions import FilePersistence, CrossPlatLock
+
+
+def validate_thread_locking_file(num):
+    with open("C:/Users/abpati/Desktop/test/lockfilelog.txt") as handle:
+        data = handle.read()
+    time_list = []
+    for line in data.split("\n"):
+        start, end = line.split("-")
+        time_list.append([start, end])
+    if len(time_list) != num:
+        print("Threads count dont match ")
+        return
+    time_list.sort()
+    prev = None
+    for interval in time_list:
+        start, end = interval
+        if start > end:
+            print("Intervals start bigger than end")
+            return
+        if prev:
+            if start < prev:
+                print("overlapping")
+        prev = start
+    print("success")
+    return
+
+
+def validate_result_in_cache(num):
+    with open("C:/Users/abpati/Desktop/test/msal.cache") as handle:
+        data = handle.read()
+    prev_tag= None
+    prev_proc_id = None
+    count = 0
+    for line in data.split(("\n")):
+        tag, proc_id = line.split(" ")
+        if tag == '<':
+            if '<' == prev_tag:
+                print("Failed")
+                return
+        elif tag == '>':
+            count+=1
+            if '<' != prev_tag or (not prev_proc_id == proc_id):
+                print("Failed")
+                return
+        else:
+            print("Unexpected Token")
+        prev_proc_id = proc_id
+        prev_tag = tag
+    if ">" != prev_tag:
+        print("Failed")
+        return
+    if count != num:
+        print("Failed")
+        return
+    print("sucess")
+
+
+def do_something(tid):
```

Comment:
`do_something` makes the code hard to read, as it is too general. Perhaps `write_to_protected_file` ?